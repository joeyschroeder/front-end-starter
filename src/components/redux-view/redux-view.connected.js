import { resetSampleValue, selectSampleValue, updateSampleValue } from 'store/sample/sample';

import { connect } from 'react-redux';
import { ReduxView } from './redux-view';

const mapStateToProps = state => ({
  value: selectSampleValue(state)
});

const mapDispatchToProps = {
  update: updateSampleValue,
  reset: resetSampleValue
};

export const ReduxViewConnected = connect(
  mapStateToProps,
  mapDispatchToProps
)(ReduxView);
