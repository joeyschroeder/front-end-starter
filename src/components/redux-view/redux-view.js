import { BlockButton } from 'components/block-button/block-button';
import PropTypes from 'prop-types';
import React from 'react';
import { Screen } from 'components/screen/screen';
import styles from './redux-view.style.scss';

export const ReduxView = props => {
  const { reset, update, value } = props;

  return (
    <Screen className={styles.root}>
      <h1>REDUX</h1>
      <p>Redux is a predictable state container for JavaScript apps.</p>
      <p>
        This application includes <strong>React-Redux</strong> along with a sample reducer and action set following the{' '}
        <a
          href="https://medium.com/@scbarrus/the-ducks-file-structure-for-redux-d63c41b7035c"
          target="_blank"
          rel="noopener noreferrer"
        >
          Redux Duck Pattern
        </a>
        .
      </p>
      <h2>Redux Value: {value}</h2>
      <p>
        Click the following buttons to update and reset the value stored in the Redux store. Note that you can leave
        this page and return and the value will be saved.
      </p>
      <BlockButton onClick={update}>Update</BlockButton>
      <BlockButton onClick={reset}>Reset</BlockButton>
    </Screen>
  );
};

ReduxView.propTypes = {
  reset: PropTypes.func,
  update: PropTypes.func,
  value: PropTypes.number
};

ReduxView.defaultProps = {
  reset: () => {},
  update: () => {},
  value: 0
};
