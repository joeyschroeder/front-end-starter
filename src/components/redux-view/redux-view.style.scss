@import "../../styles/config/variables";

.root {
  margin-left: auto;
  margin-right: auto;
  max-width: $width-mobile;
  width: 100%;
}
