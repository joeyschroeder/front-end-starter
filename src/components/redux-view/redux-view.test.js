/* global React, create */
import { ReduxView } from './redux-view';

describe('<ReduxView />', () => {
  it('should render correctly', () => {
    const component = create(<ReduxView />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
