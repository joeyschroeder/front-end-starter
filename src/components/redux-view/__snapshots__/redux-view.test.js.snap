// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReduxView /> should render correctly 1`] = `
<div
  className="animated fadeIn root root"
>
  <h1>
    REDUX
  </h1>
  <p>
    Redux is a predictable state container for JavaScript apps.
  </p>
  <p>
    This application includes 
    <strong>
      React-Redux
    </strong>
     along with a sample reducer and action set following the
     
    <a
      href="https://medium.com/@scbarrus/the-ducks-file-structure-for-redux-d63c41b7035c"
      rel="noopener noreferrer"
      target="_blank"
    >
      Redux Duck Pattern
    </a>
    .
  </p>
  <h2>
    Redux Value:
    0
  </h2>
  <p>
    Click the following buttons to update and reset the value stored in the Redux store. Note that you can leave this page and return and the value will be saved.
  </p>
  <button
    className="root"
    onClick={[Function]}
    type="button"
  >
    Update
  </button>
  <button
    className="root"
    onClick={[Function]}
    type="button"
  >
    Reset
  </button>
</div>
`;
